### PR TITLE
Create event form

### PR DESCRIPTION
**Pull request by Karvlox**
### Reviewers

@JP-Arequipa 
@CatCoder36

### CODE REVIEW

File: apps/cevent_frontend/src/app/create_event/page.tsx [hunk 1]
```diff
@@ -1,7 +1,80 @@
...
+const CreateEvent = () => {
+  const [selectedImage, setSelectedImage] = useState(null);
+  const [generalInfo, setGeneralInfo] = useState({});
+  const [dateLocation, setDateLocation] = useState({});
+  const [priceCapacity, setPriceCapacity] = useState({});
+  const [isGeneralInfoComplete, setIsGeneralInfoComplete] = useState(false);
+  const [isDateLocationComplete, setIsDateLocationComplete] = useState(false);
+  const [isPriceCapacityComplete, setIsPriceCapacityComplete] = useState(false);
+
+  const handleImageChange = (event) => {
```
Comment (CatCoder36) [inline apps/cevent_frontend/src/app/create_event/page.tsx]: Please use types in this component

**Comment by JP-Arequipa** [comment]
Fix Merge Conflicts

**Comment by JP-Arequipa** [comment]
When you open the categories section, its position is not linked to the category field
https://github.com/user-attachments/assets/b87554bb-de31-453f-8182-e931513f6522

File: apps/cevent_frontend/src/app/create_event/_components/Activities.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
+import React, { useState } from 'react';
+import { Box, Typography, TextField, Button } from '@mui/material';
+import Add from '@mui/icons-material/Add';
+import Remove from '@mui/icons-material/Remove';
+import '../_styles/Activities.css';
+
+interface Activity {
```
Comment (CatCoder36) [inline apps/cevent_frontend/src/app/create_event/_components/Activities.tsx]: Please add this interface in the utils/interfaces folder to make it accessible to all components.

File: apps/cevent_frontend/src/app/create_event/_components/Activities.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  name: string;
+  description: string;
+  startTime: string;
+  endTime: string;
+  capacity: number;
+  createdAt: string;
+}
+
+interface ActivitiesProps {
```
Comment (CatCoder36) [inline apps/cevent_frontend/src/app/create_event/_components/Activities.tsx]: The same

File: apps/cevent_frontend/src/app/create_event/_components/AditionalSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import { Box, Typography, FormControlLabel, Switch } from '@mui/material';
+import '../_styles/AditionalSettings.css';
+
+const AdditionalSettings = () => {
+  const [attendanceControl, setAttendanceControl] = useState(false);
+
+  const handleToggle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { checked } = event.target; // Desestructuramos `checked` desde `event.target`
```
Comment (CatCoder36) [inline apps/cevent_frontend/src/app/create_event/_components/AditionalSettings.tsx]: Please delete comments

File: apps/cevent_frontend/src/app/create_event/_components/DateLocation.tsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import {
+  Box,
+  Typography,
+  TextField,
+  InputAdornment,
+  Modal,
+} from '@mui/material';
+import { CalendarToday, AccessTime, LocationOn } from '@mui/icons-material';
+import Calendar from 'react-calendar'; // No es necesario importar Value
```
Comment (CatCoder36) [inline apps/cevent_frontend/src/app/create_event/_components/DateLocation.tsx]: Please delete comments

File: apps/cevent_frontend/src/app/create_event/_components/DateLocation.tsx [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+import { GoogleMap, Marker, useJsApiLoader } from '@react-google-maps/api';
+import '../_styles/DateLocation.css';
+
+type DateLocationData = {
```
Comment (CatCoder36) [inline apps/cevent_frontend/src/app/create_event/_components/DateLocation.tsx:14]: Please define a interface and move to utils/interfaces

File: apps/cevent_frontend/src/app/create_event/_components/DateLocation.tsx [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+
+      if (lat !== undefined && lng !== undefined) {
+        const newPosition = { lat, lng };
+        setMapState((prev) => ({
+          ...prev,
+          markerPosition: newPosition,
+        }));
+
+        try {
+          const response = await fetch(
```
Comment (CatCoder36) [inline apps/cevent_frontend/src/app/create_event/_components/DateLocation.tsx:73]: I could suggest to change the fetching logic to a different file and define it as a service, i.e. to have in Services a MapaService.ts that has a function GetPointInfo = (lat:number, long: number): PointInfo => {} 
All this to separate data fetching logic from UI logic.

File: apps/cevent_frontend/src/app/create_event/_components/imports.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export { default as ImageUpload } from './ImageUpload';
```
Comment (CatCoder36) [inline apps/cevent_frontend/src/app/create_event/_components/imports.js]: Use a .ts extension

**Comment by CatCoder36** [comment]
CI

File: apps/cevent_frontend/src/app/create_event/_components/GeneralInfo.tsx [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+const GeneralInfo = ({ onComplete }: { onComplete: OnCompleteCallback }) => {
+  const [fields, setFields] = useState<FieldsGeneralInfo>({
+    title: '',
+    shortDescription: '',
+    description: '',
+    categoryId: 0,
+  });
+
+  const [categories, setCategories] = useState<
+    { keyWord: string; phrase: string; color: string; id: number }[]
```
Comment (CatCoder36) [inline apps/cevent_frontend/src/app/create_event/_components/GeneralInfo.tsx:19]: Please use the intefaz EventCategory defined in util/interfaces

**Review by CatCoder36 — APPROVED** [review]

**Review by JP-Arequipa — APPROVED** [review]

**Pull request merged**